doc/filter.md

```diff
@@ -93,22 +40,43 @@ list3 = rootPackage.filterChildren(
     }
   }
 );
+```
+
+Queries
+-------
+
+The Query, introduced in Spoon 5.5 by Pavel Vojtechovsky, is an improved filter mechanism:
+
+* matching can be done with a Java 8 lambda
+* queries can be chained
 
-//creating a custom filter to select all public fields using java 8 lambda
-list3 = rootPackage.filterChildren((CtField field)->field.getModifiers.contains(ModifierKind.PUBLIC));
+`CtQueryable#filterChildren(Filter)` is a filtering query that can be chained:
 
-// a query, which processes all not deprecated methods of all deprecated classes
+```java
+// collecting all methods of deprecated classes
+list2 = rootPackage
+    .filterChildren(new AnnotationFilter(Deprecated.class))
+    .filterChildren(new TypeFilter(CtMethod.class)).list();
+```
+
+`CtQueryable#map(CtFunction)`enables you to give Java 8 lambda as query.
+A boolean return value of the lambda tells whether the elements
+are selected for inclusion or input not.
+
+```java
+// creating a custom filter to select all public fields using java 8 lambda
+list3 = rootPackage.filterChildren((CtField field)->field.getModifiers.contains(ModifierKind.PUBLIC)).list();
+```
+
+If the CtFunction returns an object, this object is given as result to the query:
+
+```java
+// a query which processes all not deprecated methods of all deprecated classes
 rootPackage
   .filterChildren((CtClass clazz)->clazz.getAnnotation(Deprecated.class)!=null)
   .map((CtClass clazz)->clazz.getMethods())
   .map((CtMethod<?> method)->method.getAnnotation(Deprecated.class)==null)
-  .forEach((CtMethod<?> method)->{
-  	... do something with found methods ...
-  });
-  
-// query which returns overridden methods of deprecated methods
-list4 = rootPackage
-  .filterChildren((CtMethod method)->method.getAnnotation(Deprecated.class)!=null)
-  .map(new OverriddenMethodFilter())
-  .list();
-```
\ No newline at end of file
+  .list()
+;
+```
+
```

Comment:
the code like this
map((CtClass clazz)->clazz.getMethods())
tests it. I did not checked how much tests are there after your simplification. I can have a look tommorrow. I plan to finish today soon.